### PR TITLE
Added json schema validation and test

### LINKED ISSUE #2: Config file for the action
### Motivation
We want to define the structure of the action config file, that the user must have created in its repository.

Default config file path: `.github/make_dispatches.yml`

```yaml
dispatches:
   # APIS
  - type: snapshots # MANDATORY
    flavors: # MANDATORY
      - default
    state_repos: # MANDATORY
      - repo: my-state-1 # (mandatory)
        base_path: apps # (optional)
        tenant: mytenant1 # (mandatory)
        application: my-app # (mandatory)
        env: dev # (mandatory)
        service_names: ['controller'] # (mandatory)
        registry: my_registry_address # (OPTIONAL: uses org SNAPSHOTS vs RELEASES)
        version: branch_dev | latest_release | latest_prerelease | release tag | commit-sha # (mandatory)
        
      - repo: my-state-repo2
        registry: my_registry_address 
        tenant: mytenant2
        application: my-app2
        env: pre
        service_names: ['controller']
        version: branch_dev | latest_release | latest_prerelease | release tag | commit-sha # (mandatory)

  # frontends
  - type: releases
    flavors:
      - flavor1
    state_repos:
      - repo: my_state_repo1
        base_path: apps
        tenant: mytenant1
        application: myapp1
        env: dev
        service_names: ['controller']
        version: branch_dev | latest_release | latest_prerelease | release tag | commit-sha # (mandatory)

  - type: releases
    flavors: 
      - flavor2
    state_repos:
      - repo: my_state2
        tenant: 
        application: myapp1
        env: pre
        service_names: ['controller']
        version: branch_dev | latest_release | latest_prerelease | release tag | commit-sha # (mandatory)
 ```

### PR DESCRIPTION
**Pull request by JulianGV21**
Closes #2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,7 +67,9 @@
   "dependencies": {
     "@actions/core": "^1.10.1",
     "@actions/github": "^6.0.0",
+    "ajv": "^8.17.1",
     "debug": "^4.3.5",
+    "js-yaml": "^4.1.0",
```
Comment (alambike) [inline package.json:72]: Already yaml dependency included

File: schema/jsonschema.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+              },
+              "required": [
+                "repo",
+                "tenant",
+                "application",
+                "env",
+                "service_names",
+                "version"
+              ],
+              "additionalProperties": true
```
Comment (alambike) [inline schema/jsonschema.json:69]:
```suggestion
              "additionalProperties": false
```

All properties should be controlled by the json schema, not additional properties allowed

File: validation.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+const validate = ajv.compile(schema)
+
+const valid = validate(yamlData)
+
+if (valid) {
+  console.log('Successful validation')
+} else {
+  console.error('Validation error:', validate.errors)
+  process.exit(1)
+}
```
Comment (alambike) [inline validation.js:26]: This is a test file, you don't need to do a process exit!

We are using Jest to create the tests cases: https://jestjs.io/es-ES/

You must define a tests case and use `expect` to validate the expected results, and the tests will pass or fail according to the results of the expected evaluation. 

See for example: https://github.com/prefapp/action-make-state-repos-dispatches/blob/f7a81d0ed6282a95f0c83f485789050aa701a057/__tests__/dispatcher.test.js#L75-L84

**Review by alambike — CHANGES_REQUESTED** [review]

**Review by alambike — APPROVED** [review]

File: schema/jsonschema.json
Comment (alambike) [inline schema/jsonschema.json:1]: Looks good but it needs to be used when loading the config file: https://github.com/prefapp/action-make-state-repos-dispatches/blob/main/utils/text-helper.js

**Review by alambike — CHANGES_REQUESTED** [review]

File: utils/validate.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+const YAML = require('yaml')
+const fs = require('fs')
+const path = require('path')
+const Ajv = require('ajv')
+
+function ConfigParse(fileContent, encoding = '') {
```
Comment (alambike) [inline utils/validate.js:6]:
```suggestion
function configParse(fileContent, encoding = '') {
```

File: utils/validate.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    return yamlData
+  } catch (err) {
+    throw new Error(`Error parsing YAML file: ${err.message}`)
+  }
+}
+
+module.exports = {
+  ConfigParse
```
Comment (alambike) [inline utils/validate.js:34]:
```suggestion
  configParse
```

File: validation.test.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
+const path = require('path')
+const fs = require('fs')
+const yaml = require('yaml')
+const { ConfigParse } = require('./utils/validate')
```
Comment (alambike) [inline validation.test.js:4]:
```suggestion
const { configParse } = require('./utils/validate')
```

File: validation.test.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    yamlWithoutRequiredField.dispatches[0].state_repos =
+      yamlWithoutRequiredField.dispatches[0].state_repos.map(repo => {
+        const { repo: removedRepo, ...rest } = repo
+        return rest
+      })
+
+    expect(() =>
+      ConfigParse(yaml.stringify(yamlWithoutRequiredField))
```
Comment (alambike) [inline validation.test.js:47]:
```suggestion
      configParse(yaml.stringify(yamlWithoutRequiredField))
```

File: utils/validate.js (hunk 4 above)
Comment (alambike) [reply utils/validate.js:6]: Please, take a look to the style guide we use to follow: https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers

**Review by alambike — APPROVED** [review]

**Review by juanjosevazquezgil — APPROVED** [review]

**Pull request merged**